bot/exts/evergreen/wikipedia.py

```diff
@@ -37,37 +38,42 @@ async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List
         """Search wikipedia search string and return formatted first 10 pages found."""
         url = SEARCH_API.format(number_of_results=10, string=search)
         async with self.bot.http_session.get(url=url) as resp:
-            if resp.status == 200:
-                raw_data = await resp.json()
-                number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
+            if resp.status != 200:
+                log.info(f"Unexpected response `{resp.status}` while searching wikipedia for `{search}`")
+                raise ExternalAPIError("Wikipedia API")
 
+            raw_data = await resp.json()
+            if raw_data.get("query", None) is None:
+                if raw_data.get("errors", None) is not None:
```

Comment:
`dict.get()` sets the second parameter i.e. the default value as `None` by default so that can be ignored, another thing you can change is `is None` and `is not none` using `not`.

```suggestion
            if not raw_data.get("query"):
                if raw_data.get("errors"):
```